genshin/client/components/hoyolab.py

```diff
@@ -29,6 +29,25 @@ async def search_users(
         )
         return [models.PartialHoyolabUser(**i["user"]) for i in data["list"]]
 
+    async def get_hoyolab_self(self, *, lang: typing.Optional[str] = None) -> models.PartialHoyolabUser:
+        """Get hoyolab data for me."""
+        if self.region == types.Region.OVERSEAS:
+            url = "https://bbs-api-os.hoyolab.com/community/painter/wapi/user/full"
+            referer = "https://www.hoyolab.com/"
+        elif self.region == types.Region.CHINESE:
+            url = "https://bbs-api.mihoyo.com/user/wapi/getUserFullInfo?gids=2"
+            referer = "https://bbs.mihoyo.com/"
+        else:
+            raise TypeError(f"{self.region!r} is not a valid region.")
+
+        data = await self.request_hoyolab(
+            url=url,
+            lang=lang,
+            headers={"Referer": referer},
```

Comment:
Referer is never actually needed afaik.

Reply:
No, in my test, referer is strictly required, otherwise the endpoint returns forbidden.

Reply:
maybe the referer is checked only in https://bbs-api-os.hoyolab.com and https://bbs-api.mihoyo.com, not checked in TAKUMI_URL api?

Reply:
Considering many other libraries add Referer everywhere it might not be a bad idea to do the same. I already provide user agents anyways.

Reply:
Now new request_bbs apply referer automatically.